tests/ijby/test-cut.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#-------------------------------------------------------------------------------
+# Copyright 2020 H2O.ai
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+from datatable import dt, stype, f, cut
+from tests import assert_equals
+import pytest
+import random
+import math
+
+
+#-------------------------------------------------------------------------------
+# cut()
+#-------------------------------------------------------------------------------
+
+def test_cut_error_noargs():
+    msg = r"Function cut\(\) requires one positional argument, but none were given"
+    with pytest.raises(TypeError, match=msg):
+        cut()
+
+
+def test_cut_error_wrong_column_types():
+    DT = dt.Frame([[1, 0], ["1", "0"]])
+    msg = r"cut\(\) can only be applied to numeric columns, instead column 1 has an stype: str32"
+    with pytest.raises(TypeError, match=msg):
+        cut(DT)
+
+
+def test_cut_error_negative_bins():
+    msg = "Number of bins must be positive, instead got: -10"
+    DT = dt.Frame(range(10))
+    with pytest.raises(ValueError, match=msg):
+        cut(DT, -10)
```

Comment:
Also: add test for 0 bins

Reply:
I guess I had it, but it got somehow removed in one of the re-factoring iterations. WIll add some more.